sympy/functions/elementary/trigonometric.py

```diff
@@ -117,10 +117,19 @@ def _peeloff_pi(arg):
             K, p = a.as_two_terms()
             if p is S.Pi and K.is_Rational:
                 break
+            K = a.coeff(S.Pi)
+            if K and K.is_rational:
```

Comment:
However I think here `K.is_rational` was sufficient. What makes to check `if K`. Any particular reason?

Reply:
Actually i was also of the same opinion but since @smichr had written it this way, I didn't bother to change it.

Reply:
Simply K.is_rational is working, so i have made the changes

Reply:
What happens if there is no pi? I think that's why coeff would return 0.

Reply:
Nothing to worry still the result returned would be arg,zero
As then m2 will be zero